haystack/tools/toolset.py

```diff
@@ -289,3 +295,68 @@ def __getitem__(self, index):
         :returns: The Tool at the specified index
         """
         return self.tools[index]
+
+
+class _ToolsetWrapper(Toolset):
```

Comment:
could you explain why we need `_ToolsetWrapper`?

Reply:
Not sure we need it in this form (there should be something simpler), it was a hack to enable:
```
all_tools = perplexity_mcp + routing_mcp + some_other_tools
```

If they are all MCPToolset and not warmed up when addition above is executed (with current code) we'll just get garbage. We need to preserve their configs (URLs) and then after they've been warmed up they could be added  (using our current code) - this way we effectively add them and preserve configs so that when we warm them up they all connect to the right servers.